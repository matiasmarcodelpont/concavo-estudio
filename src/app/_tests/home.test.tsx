import '@testing-library/jest-dom'
import { render, screen, within } from '@testing-library/react'

import Home from '../page'

jest.mock('@/data/data.tsx', () => ({
  productos: [
    {
      slug: 'azulejos-los-peores',
      name: 'Azulejos los peores',
      colaborador: {
        slug: 'facu',
      },
    },
    {
      slug: 'azulejos-verdes',
      name: 'Azulejos verdes',
      colaborador: null,
    },
    {
      slug: 'azulejos-los-mejores',
      name: 'Azulejos los mejores',
      colaborador: {
        slug: 'mati',
      },
    },
    {
      slug: 'azulejos-azules',
      name: 'Azulejos azules',
      colaborador: null,
    },
  ],
}))

describe('Home', () => {
  it('renders homepage unchanged', () => {
    const { container } = render(<Home />)
    expect(container).toMatchSnapshot()
  })

  it('renders a video with a link', () => {
    const component = render(<Home />)
    const video = component.getByTitle('Casa CÃ³ncavo')

    const link = within(video.parentElement as HTMLElement).getByRole<HTMLAnchorElement>('link')
    expect(link.href).toMatch(/^https?:\/\/[^/]+\/casa-concavo\/white-haven$/)
  })

  it('renders concavo productos', () => {
    render(<Home />)

    const productosList = screen.getByRole('list', { name: 'Nuestra tienda' })
    expect(productosList).toBeInTheDocument()

    const productos = within(productosList).getAllByRole('listitem')
    expect(productos).toHaveLength(2)
  })
})
