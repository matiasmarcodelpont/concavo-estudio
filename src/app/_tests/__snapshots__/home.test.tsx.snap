// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
<div>
  <main
    class="text-center"
  >
    <div
      class="h-[17vh]"
    />
    <h1
      class="animate-fade-in delay-250 sticky top-4 z-30 mb-4 my-0"
    >
      <img
        alt="Cóncavo"
        class="object-contain w-[150px] sm:w-[200px] m-auto"
        data-nimg="1"
        decoding="async"
        height="176"
        loading="lazy"
        src="/_next/image?url=%2Flogos%2Fconcavo_black.png&w=1920&q=75"
        srcset="/_next/image?url=%2Flogos%2Fconcavo_black.png&w=750&q=75 1x, /_next/image?url=%2Flogos%2Fconcavo_black.png&w=1920&q=75 2x"
        style="color: transparent;"
        width="690"
      />
    </h1>
    <section
      class="animate-fade-in delay-250 flex flex-col justify-center items-center"
    >
      <p
        class="font-editorial-new font-extralight sm:text-xl mb-6"
      >
        Un camino de disfrute y creación.
        <br />
        De imaginar espacios relajados, armónicos y luminosos.
        <br />
        De crear sensaciones de 
        <em>
          calma.
        </em>
        <br />
      </p>
      <svg
        class="text-black size-12"
        fill="none"
        height="24"
        viewBox="0 0 569 569"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <mask
          height="569"
          id="mask0_2090_30"
          maskUnits="userSpaceOnUse"
          style="mask-type: alpha;"
          width="569"
          x="0"
          y="0"
        >
          <rect
            fill="currentColor"
            height="569"
            width="569"
          />
        </mask>
        <g
          mask="url(#mask0_2090_30)"
        >
          <mask
            height="356"
            id="mask1_2090_30"
            maskUnits="userSpaceOnUse"
            style="mask-type: luminance;"
            width="569"
            x="1"
            y="107"
          >
            <path
              d="M569.616 107H1V462.552H569.616V107Z"
              fill="white"
            />
          </mask>
          <g
            mask="url(#mask1_2090_30)"
          >
            <mask
              height="118"
              id="mask2_2090_30"
              maskUnits="userSpaceOnUse"
              style="mask-type: luminance;"
              width="282"
              x="144"
              y="109"
            >
              <path
                d="M144.454 109.179H425.761V226.542H144.454V109.179Z"
                fill="white"
              />
            </mask>
            <g
              mask="url(#mask2_2090_30)"
            >
              <mask
                height="118"
                id="mask3_2090_30"
                maskUnits="userSpaceOnUse"
                style="mask-type: luminance;"
                width="282"
                x="144"
                y="109"
              >
                <path
                  d="M425.688 109.179C425.688 173.863 362.762 226.303 285.146 226.303C207.525 226.303 144.6 173.863 144.6 109.179"
                  fill="white"
                />
              </mask>
              <g
                mask="url(#mask3_2090_30)"
              >
                <path
                  d="M568.066 462.375H2.14941V107.343H568.066V462.375Z"
                  fill="currentColor"
                />
              </g>
            </g>
            <mask
              height="235"
              id="mask4_2090_30"
              maskUnits="userSpaceOnUse"
              style="mask-type: luminance;"
              width="564"
              x="3"
              y="226"
            >
              <path
                d="M3.98535 226.178H566.235V460.538H3.98535V226.178Z"
                fill="white"
              />
            </mask>
            <g
              mask="url(#mask4_2090_30)"
            >
              <mask
                height="235"
                id="mask5_2090_30"
                maskUnits="userSpaceOnUse"
                style="mask-type: luminance;"
                width="563"
                x="4"
                y="226"
              >
                <path
                  d="M566.23 226.302C566.23 355.665 440.383 460.538 285.147 460.538C129.905 460.538 4.05859 355.665 4.05859 226.302"
                  fill="white"
                />
              </mask>
              <g
                mask="url(#mask5_2090_30)"
              >
                <path
                  d="M568.066 462.374H2.14941V107.343H568.066V462.374Z"
                  fill="currentColor"
                />
              </g>
            </g>
          </g>
        </g>
      </svg>
    </section>
    <div
      class="h-[17vh]"
    />
    <section
      class="relative h-[60vw] min-h-[400px] max-h-[80vh] group mb-12"
    >
      <video
        autoplay=""
        class="absolute top-0 left-0 w-full h-full pointer-events-none -z-10 border-none object-cover"
        height="100%"
        loop=""
        src="/casa.mp4"
        title="Casa Cóncavo"
      />
      <div
        class="flex flex-col h-full bg-[#0003] justify-around items-center"
      >
        <div
          class="pt-2 pb-1 border-white uppercase invisible"
        >
          Start here
        </div>
        <p
          class="text-bone text-xl sm:text-2xl font-bold uppercase tracking-[.2em]"
        >
          Casa
          <img
            alt="Cóncavo"
            class="object-contain w-[300px] sm:w-[400px] -mt-[.5em]"
            data-nimg="1"
            decoding="async"
            height="3715"
            loading="lazy"
            src="/_next/image?url=%2Flogos%2Fconcavo_white.png&w=3840&q=75"
            srcset="/_next/image?url=%2Flogos%2Fconcavo_white.png&w=3840&q=75 1x"
            style="color: transparent;"
            width="12773"
          />
        </p>
        <a
          class="font-editorial-new px-6 pt-2 pb-1 text-bone border rounded-full border-white uppercase"
          href="/casa-concavo"
        >
          Start here
        </a>
      </div>
    </section>
    <section
      class="mx-6 sm:mx-10 md:mx-12 mb-12"
    >
      <span
        style="font-size: 0px;"
      />
      <h1
        class="font-editorial-new tracking-[0.01em] uppercase mt-12 mb-6 sm:mt-16 sm:mb-9 md:mt-20 md:mb-12 text-2xl md:text-3xl opacity-0"
        id="productos-concavo"
      >
        Nuestra tienda
      </h1>
      <ul
        aria-labelledby="productos-concavo"
        class="grid grid-cols-auto-fit-100 xs:grid-cols-auto-fit-150 sm:grid-cols-auto-fill-200 md:grid-cols-auto-fill-300 gap-x-2 gap-y-8 md:gap-y-10"
      >
        <li>
          <a
            aria-label="Azulejos verdes"
            class="cursor-pointer text-dark-gray group"
            href="/productos/azulejos-verdes"
          >
            <span
              style="font-size: 0px;"
            />
            <div
              class="pt-[100%] relative overflow-hidden "
            >
              <img
                alt="Azulejos verdes"
                class="object-cover group-hover:scale-105 transition-all duration-1000 opacity-0"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                sizes="100vw"
                src="/_next/image?url=%2Fproductos%2Fconcavo%2Fazulejos-verdes.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fproductos%2Fconcavo%2Fazulejos-verdes.jpg&w=640&q=75 640w, /_next/image?url=%2Fproductos%2Fconcavo%2Fazulejos-verdes.jpg&w=750&q=75 750w, /_next/image?url=%2Fproductos%2Fconcavo%2Fazulejos-verdes.jpg&w=828&q=75 828w, /_next/image?url=%2Fproductos%2Fconcavo%2Fazulejos-verdes.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fproductos%2Fconcavo%2Fazulejos-verdes.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fproductos%2Fconcavo%2Fazulejos-verdes.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fproductos%2Fconcavo%2Fazulejos-verdes.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fproductos%2Fconcavo%2Fazulejos-verdes.jpg&w=3840&q=75 3840w"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
              />
            </div>
            <div
              class="group-hover:text-light-gray transition-colors"
            >
              <span
                style="font-size: 0px;"
              />
              <p
                class="font-editorial-new tracking-[0.01em] uppercase opacity-100 text-sm xs:text-base sm:text-lg md:text-xl mt-3 sm:mt-3 md:mt-5 mb-0 sm:mb-0 md:mb-0"
              >
                Azulejos verdes
              </p>
              <p
                aria-label="dimensions"
                class="text-xs text-center mt-0 leading-none"
              >
                AAA x LLL x PPP
              </p>
            </div>
          </a>
        </li>
        <li>
          <a
            aria-label="Azulejos azules"
            class="cursor-pointer text-dark-gray group"
            href="/productos/azulejos-azules"
          >
            <span
              style="font-size: 0px;"
            />
            <div
              class="pt-[100%] relative overflow-hidden "
            >
              <img
                alt="Azulejos azules"
                class="object-cover group-hover:scale-105 transition-all duration-1000 opacity-0"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                sizes="100vw"
                src="/_next/image?url=%2Fproductos%2Fconcavo%2Fazulejos-azules.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fproductos%2Fconcavo%2Fazulejos-azules.jpg&w=640&q=75 640w, /_next/image?url=%2Fproductos%2Fconcavo%2Fazulejos-azules.jpg&w=750&q=75 750w, /_next/image?url=%2Fproductos%2Fconcavo%2Fazulejos-azules.jpg&w=828&q=75 828w, /_next/image?url=%2Fproductos%2Fconcavo%2Fazulejos-azules.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fproductos%2Fconcavo%2Fazulejos-azules.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fproductos%2Fconcavo%2Fazulejos-azules.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fproductos%2Fconcavo%2Fazulejos-azules.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fproductos%2Fconcavo%2Fazulejos-azules.jpg&w=3840&q=75 3840w"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
              />
            </div>
            <div
              class="group-hover:text-light-gray transition-colors"
            >
              <span
                style="font-size: 0px;"
              />
              <p
                class="font-editorial-new tracking-[0.01em] uppercase opacity-100 text-sm xs:text-base sm:text-lg md:text-xl mt-3 sm:mt-3 md:mt-5 mb-0 sm:mb-0 md:mb-0"
              >
                Azulejos azules
              </p>
              <p
                aria-label="dimensions"
                class="text-xs text-center mt-0 leading-none"
              >
                AAA x LLL x PPP
              </p>
            </div>
          </a>
        </li>
      </ul>
    </section>
  </main>
</div>
`;
